### ISSUE TITLE
instanceof not respected

### ISSUE THREAD

**Issue by orklah** [vimeo/psalm#7705]
Something wrong going on here: https://psalm.dev/r/3d69cb785e

But I don't have time to check it for now

**Comment by weirdan** [comment]
It boils down to two `Code` parts (`Code|enum(Code::Ok)`) being present in the union. One is removed by the instanceof check (`enum(Code::Ok)`), but the other remains.

**Comment by orklah** [comment]
yeah, when I checked, it seemed to be that TNamedObject wasn't filtered because it's dangerous to filter one type based on instanceof (we have a test like that where `if($aThrowable instanceof $anotherThrowable)` can be false, but I couldn't understand why it's filtered when the enum is not there...
